### PR TITLE
Update to 1.17

### PR DESCRIPTION
**Pull request by BeetMacol**
Updates the mod to **Minecraft 1.17.1**. Also updates to Java 16, but I didn't add any Java 9+ features here.

### CODE REVIEW

**Comment by BeetMacol** [comment]
I'm not sure, but it seems like this mirror is not updated with the GitLab repo. f0ab32f3c8c56c4c793ab34d2167e4b58cf4bf62 was pushed by @YTG1234 and is on the GitLab remote, but didn't end up here (and that's why this pull request also adds the YTG's commit).

**Comment by YTG1234** [comment]
> I'm not sure, but it seems like this mirror is not updated with the GitLab repo. f0ab32f3c8c56c4c793ab34d2167e4b58cf4bf62 was pushed by @YTG1234 and is on the GitLab remote, but didn't end up here (and that's why this pull request also adds the YTG's commit).

That's weird. I set up the GL repo to automatically mirror.

On that note, I think I forgot to mention anywhere that this is an effectively read-only mirror, and only syncs GL->GH. I can (and probably will) merge this manually, but I would prefer that further PRs be opened on GitLab instead.

After all PRs are merged/closed, I will disable PRs in this repo.

File: build.gradle.kts [hunk 1]
```diff
@@ -52,8 +53,8 @@ allprojects {
     }
 
     java {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_16
+        targetCompatibility = JavaVersion.VERSION_16
     }
```
Comment (YTG1234) [inline build.gradle.kts]: This is now unnecessary as `release` is set.

**Review by YTG1234 — CHANGES_REQUESTED** [review]
The mixins are in `remappedSrc`, it seems.

**Comment by BeetMacol** [comment]
Done, fixed those issues.

> On that note, I think I forgot to mention anywhere that this is an effectively read-only mirror, and only syncs GL->GH. I can (and probably will) merge this manually, but I would prefer that further PRs be opened on GitLab instead.

I thought it's a both way mirror. I can make a GL account and *merge request* there if it's a big trouble for you.

**Comment by YTG1234** [comment]
> I thought it's a both way mirror.

I would have made it so, but it's apparently unstable and unsupported.

**Pull request merged**